org.eclipse.dawnsci.nexus.test/src/org/eclipse/dawnsci/nexus/file/NexusFileRemoveNodeTest.java

```diff
@@ -0,0 +1,83 @@
+package org.eclipse.dawnsci.nexus.file;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+import java.io.File;
+
+import org.eclipse.dawnsci.analysis.api.tree.DataNode;
+import org.eclipse.dawnsci.analysis.api.tree.GroupNode;
+import org.eclipse.dawnsci.nexus.NexusException;
+import org.eclipse.dawnsci.nexus.NexusFile;
+import org.eclipse.dawnsci.nexus.TestUtils;
+import org.eclipse.dawnsci.nexus.test.util.NexusTestUtils;
+import org.eclipse.january.dataset.Dataset;
+import org.eclipse.january.dataset.DatasetFactory;
+import org.eclipse.january.dataset.DatasetUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class NexusFileRemoveNodeTest {
+
+	private static String testScratchDirectoryName;
+	private static String FILE_NAME;
+
+	/**
+	 * Creates an empty directory for use by test code.
+	 *
+	 * @throws Exception if setup fails
+	 */
+	@BeforeClass
+	public static void setUpBeforeClass() throws Exception {
+		testScratchDirectoryName = TestUtils.generateDirectorynameFromClassname(NexusFileTest.class.getCanonicalName());
+		TestUtils.makeScratchDirectory(testScratchDirectoryName);
+		FILE_NAME = testScratchDirectoryName + "test.nxs";
+	}
+
+	private NexusFile setUpFile(boolean swmr) throws NexusException {
+		String fileName = swmr ? "swmr_" + FILE_NAME : FILE_NAME;
+		new File(fileName).delete();
+		NexusFile nf = NexusTestUtils.createNexusFile(fileName, swmr);
+		GroupNode g = nf.getGroup("/entry/blah", true);
+		nf.createData(g, "heros", DatasetFactory.ones(2,3,4,5));
+		if (swmr) {
+			nf.activateSwmrMode();
+		}
+		return nf;
+	}
+
+	private void removeNode(boolean swmr) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)